### PR TITLE
pytest works now

### PR DESCRIPTION
**Pull request by dataprob**
- Fixed import error for each python test file.
- Match what's been done in cpp testing, BaseTest is now called boilerplate.
- Refactored boilerplate, data_structures, and helpers.
- Inside boilerplate, `class BaseTest` is called `class BoilerTest`. I wanted to name it `Test`, but pytest will give warning to any class with name that starts with `Test`.

Screenshot 2023-06-01 at 8 52 10 PM

### CODE REVIEW

**Comment by PhaseRush** [comment]
Can you get pytest to also show the runtime for each test?

**Comment by PhaseRush** [comment]
Could we fix the directories so we don't need to hack sys every time?

File: test/python/python_deps/data_structures.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from typing import List, Optional
+
+
+class ListNode:
+    def __init__(self, val=0, next=None):
+        self.val = val
+        self.next = next
+
+
+class Helpers:
```
Comment (PhaseRush) [inline test/python/python_deps/data_structures.py:10]: Does this need to be in a class? It's already in the ds file so the helper is already implied via filename and function name.

**Comment by PhaseRush** [comment]
> Can you get pytest to also show the runtime for each test?

https://stackoverflow.com/a/27899853

Can you check if that works? Might be a easy solution.

**Comment by PhaseRush** [comment]
Do you think it's ok to stay kinda scrappy or should we follow best practices and make a venv? I've previously kept the code light, but if you plan on removing the main methods from tests, we need a way to give new envs a setup. 

https://docs.pytest.org/en/7.1.x/explanation/goodpractices.html

**Comment by PhaseRush** [comment]
Nice work @dataprob, merge when ready!

**Comment by dataprob** [comment]
finding alternative pytest duration solus now, bc tried stackoverflow, didn't work as I expected

**Comment by PhaseRush** [comment]
If pytest allows for stdout, can we print it ourselves? Does pytest support any interceptors?

**Comment by dataprob** [comment]
https://pypi.org/project/pytest-durations/ is too advanced right?

**Comment by PhaseRush** [comment]
Your judgement call to make! I don't find it to be too advanced if we're going to have dependencies anyways, but I can support either argument.

**Comment by dataprob** [comment]
In the iteration when I still had your timer code, pytest only showed its pytest time, and didn't show your timer output (or any print statements)

**Comment by PhaseRush** [comment]
There seem to be some options here

https://docs.pytest.org/en/7.1.x/how-to/capture-stdout-stderr.html

**Comment by dataprob** [comment]
I can `pytest -s` get stdout, so I'm bring timer back now!

**Comment by dataprob** [comment]
run all
Screenshot 2023-06-01 at 9 45 51 PM

run individual
Screenshot 2023-06-01 at 9 45 36 PM

**Comment by dataprob** [comment]
> * Add instructions in `run.md` on how to run all python tests

blank @Person314159 this is fixed!

**Comment by dataprob** [comment]
> * Running the python tests directly from the root dir or the `test/python` dir doesn't work

Yeah, pytest and python has their own import _hacks_. We prob can only choose one or the other.
@Person314159

**Pull request merged**